Update default limits for the joint limits and safety limits (#249)
ros/urdfdom: README.md

- **Deceleration limit**: The `deceleration` attribute specifies the maximum joint deceleration
- **Jerk limit**: The `jerk` attribute specifies the maximum joint jerk (rate of change of acceleration)

Version 1.2 also relaxes some requirements on the existing joint limit attributes:

> - `lower` and `upper` are optional for non-`revolute`/non-`prismatic` joints and default to `NaN` when omitted

Reviewer: cross-posting my comment from https://github.com/ros/urdfdom_headers/pull/95#pullrequestreview-4079189997: how do you feel about default values of `-Inf` for lower and `+Inf` for upper instead of `NaN`? this would match the default values used in SDFormat

Author: Copy that

Author: Done 👍🏾

> - `lower` and `upper` are optional for non-`revolute`/non-`prismatic` joints and default to `-infinity` and `infinity` when omitted
> - for `revolute` and `prismatic` joints, both `lower` and `upper` must be provided

Reviewer 2: ```suggestion
- `lower` and `upper` are optional and default to `-infinity` and `infinity` respectively when omitted
```

Author: Same as below

Reviewer: I think the text is fine as-is

| `deceleration` | Maximum joint deceleration (rad/s² or m/s²) | Same as `acceleration`, or infinity if `acceleration` is not set | Must be non-negative |
| `jerk` | Maximum joint jerk (rad/s³ or m/s³) | Infinity | Must be non-negative |

#### Joint Limit Semantics By Version

The parser keeps legacy behavior for older URDF versions and only applies the relaxed defaults starting in version 1.2.

| Attribute | Version 1.0 / 1.1 | Version 1.2 |
|-----------|-------------------|-------------|

> | `lower` | Optional, defaults to `0.0` when omitted | Optional for non-revolute/non-prismatic joints, defaults to `-infinity` when omitted |
> | `upper` | Optional, defaults to `0.0` when omitted | Optional for non-revolute/non-prismatic joints, defaults to `infinity` when omitted |

Reviewer 2: ```suggestion
| `lower` | Optional, defaults to `0.0` when omitted | Optional, defaults to `-infinity` when omitted |
| `upper` | Optional, defaults to `0.0` when omitted | Optional, defaults to `infinity` when omitted |
```

Author: Same as above

Reviewer: I think the text is fine as-is
